src/docfx/restore/Restore.cs

```diff
@@ -81,5 +75,115 @@ private static async Task<bool> RestoreDocset(string docsetPath, string outputPa
                 return false;
             }
         }
+
+        private static async Task RestoreFiles(string docsetPath, Config config, ErrorLog errorLog)
+        {
+            var credentialProvider = config.GetCredentialProvider();
+            var fileResolver = new FileResolver(docsetPath, credentialProvider, new OpsConfigAdapter(errorLog, credentialProvider));
+            await ParallelUtility.ForEach(config.GetFileReferences(), fileResolver.Download);
+        }
+
+        private static void RestorePackages(string docsetPath, Config config, string locale, Repository repository)
+        {
+            using (var packageResolver = new PackageResolver(docsetPath, config))
+            {
+                ParallelUtility.ForEach(
+                    GetPackages(config, locale, repository).Distinct(),
+                    item => packageResolver.DownloadPackage(item.package, item.flags),
+                    Progress.Update,
+                    maxDegreeOfParallelism: 8);
```

Comment:
should this be related to `Environment.ProcessorCount` instead of a fixed number like `ParallelUtility`?

Reply:
https://github.com/dotnet/docfx/pull/5009